contracts/router/modules/pool/traderjoe/TraderJoeV21Module.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {ILBRouter} from "../../../interfaces/traderjoe/ILBRouter.sol";
+import {ILBPair} from "../../../interfaces/traderjoe/ILBPair.sol";
+import {IndexedToken, IPoolModule} from "../../../interfaces/IPoolModule.sol";
+
+import {TraderJoeModule} from "./TraderJoeModule.sol";
+
+contract TraderJoeV21Module is TraderJoeModule {
+    constructor(address _lbRouter) TraderJoeModule(_lbRouter) {}
+
+    function version() public view override returns (ILBRouter.Version) {
+        return ILBRouter.Version.V2_1;
+    }
+
+    function _binStep(address pool) internal view override returns (uint256) {
+        return uint256(ILBPair(pool).getBinStep());
+    }
+
+    /// @inheritdoc IPoolModule
+    function getPoolQuote(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn,
+        bool probePaused
+    ) public view override returns (uint256 amountOut) {
+        address[] memory tokens = getPoolTokens(pool);
+        bool swapForY = (tokenTo.token == tokens[1]);
```

Comment:
With `IERC20` replaced by `address` this could be simplified to following:
```suggestion
        bool swapForY = (tokenTo.token == ILBPair(pool).getTokenY());
```
Which should be a bit more effective, and a bit more clear imo.